pkg/kv/iterators_test.go

```diff
@@ -0,0 +1,29 @@
+package kv_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/kv/mock"
+)
+
+func TestPartitionIterator_ClosedBehaviour(t *testing.T) {
```

Comment:
If I understand correctly - this test doesn't reproduce the original issue?

Reply:
Wanted test what will happen when getting an error inside SeekGE calling to store's Scan func.
Wanted to address the following:
- When calling close we check the flag and not the pointer.
- Close mark that iterator was closed - prevent crash in calling multiple times to Close.
- Err implementation return the iterator's error based on flag.